### ISSUE TITLE
App Stuck on relaunching first time from icon in Android

### ISSUE THREAD

**Issue by haris008** [react-native-component/react-native-smart-splash-screen#53]
Step 1 : Install app - Splash shows and close as expected
Step 2 : put app in background by pressing home button
Step : open app from launcher icon, it is stucked at splash screen.

After closing app, it works okay. Only happens on fresh install.

**Comment by bhatti-waqas** [comment]
@haris008 : Please use launch mode as "Single Task" in main activity of android manifest. This issue will be gone.

**Comment by tamirwd** [comment]
@bhatti-waqas I have the same issue, but when clicking on android Back button in step 2 above, tried you solution and unfortunately it didn't work for me. Is there something else I can try to support also this case ?

**Comment by bhatti-waqas** [comment]
@tamirwd: Can you please make sure closing splash screen code i.e 
`SplashScreen.close({
          animationType: SplashScreen.animationType.scale,
          duration: 850,
          delay: 500,
      });`
is getting executed when you re-run the app ?

**Comment by tamirwd** [comment]
@bhatti-waqas actually it's not. so I need to detect that the app is re-opened and call this method ? but I thought the splash screen should not even appear in this case no ?

**Comment by bhatti-waqas** [comment]
@tamirwd: Yes, you are right. actually I am not very good with android life cycle. I am an ios developer. 
So in my case splash screen was re-appearing when opening app again. And I made sure I close it with the code provided. In my landing screen.

**Comment by tamirwd** [comment]
@bhatti-waqas ok, I used AppState.addEventListener to detect when app is resumed, then I call SplashScreen.close again and it works ! thanks.

**Comment by Chetan8488** [comment]
@Chetan8488 : put below code in onCreate() method of SplashActivity.

if (!isTaskRoot()
                && getIntent().hasCategory(Intent.CATEGORY_LAUNCHER)
                && getIntent().getAction() != null
                && getIntent().getAction().equals(Intent.ACTION_MAIN)) {

            finish();
            return;
        }

**Comment by vitto-moz** [comment]
@tamirwd , can you, please, provide more details about your solution? Thanks
